hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapPutMessageTask.java

```diff
@@ -36,17 +38,17 @@ protected AbstractMapPutMessageTask(ClientMessage clientMessage, Node node, Conn
 
     @Override
     protected void beforeProcess() {
-        startTime = System.currentTimeMillis();
+        startTime = System.nanoTime();
     }
 
     @Override
     protected void beforeResponse() {
-        final long latency = System.currentTimeMillis() - startTime;
+        final long latency = System.nanoTime() - startTime;
         final MapService mapService = getService(MapService.SERVICE_NAME);
         MapContainer mapContainer = mapService.getMapServiceContext().getMapContainer(getDistributedObjectName());
         if (mapContainer.getMapConfig().isStatisticsEnabled()) {
             mapService.getMapServiceContext().getLocalMapStatsProvider().getLocalMapStatsImpl(getDistributedObjectName())
-                    .incrementPuts(latency);
+                    .incrementPuts(NANOSECONDS.toMicros(latency));
```

Comment:
I would be `incrementPutLatencyMicros`, but I would prefer to rename the parameters, not the method name:
`incrementPuts(long latency)` -> `incrementPuts(long latencyMicros)`
`incrementPuts(long delta, long latency)` -> `incrementPuts(long delta, long latencyMicros)`
etc.